lib/helper.js

```diff
@@ -256,18 +289,14 @@ Helper.prototype.waitForUrl = function(url) {
   var self = this;
   var filter;
   if(typeof url === 'function') {
-    filter = url;
+    filter = function() {return browser.driver.getCurrentUrl().then(url);};
   } else {
     if(url.indexOf('http') !== 0) {
       url = this.baseUrl + url;
     }
-    filter = function(currentUrl) {
-      return currentUrl === url;
-    };
+    filter = EC.urlIs(url);
   }
-  return browser.driver.wait(function() {
-    return browser.driver.getCurrentUrl().then(filter);
-  }, self.MAX_TIMEOUT, 'should wait for url: ' + url);
+  return browser.wait(filter, self.MAX_TIMEOUT, 'should wait for url: ' + url);
```

Comment:
This print out may be a bit goofy when `url` is a function, but that's how it was in the old code as well; just commenting.